lib/kontena/actors/etcd_watcher.rb

```diff
@@ -0,0 +1,52 @@
+require 'concurrent/timer_task'
+require 'etcd'
+
+module Kontena::Actors
+  class EtcdWatcher < Concurrent::Actor::RestartingContext
+    include Kontena::Logging
+
+    attr_reader :client, :path
+
+    def initialize(host, path)
+      info "initialized with etcd host=#{host} path=#{path}"
+      @host = host
+      @path = path
+      @read_task = Concurrent::TimerTask.new(execution_interval: 1) {
```

Comment:
Couldn't we use also etcd watch with correct path for this?

Reply:
The [`Kontena::Etcd::Reader`](https://github.com/SpComb/kontena-etcd/blob/master/lib/kontena/etcd/reader.rb) is intended to provide that functionality.